File: leetcode-solutions/0485-max-consecutive-ones/solution.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int maxCount = 0;
        int count = 0;

        for (int i : nums){
            if (i == 1){
                count = count + 1;
                maxCount = max(maxCount, count);
            }
            else{
                count = 0;
            }
        }

        return maxCount;
    }
};

File: leetcode-solutions/0485-max-consecutive-ones/solution.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        l = 0
        count = 0
        maxCount = 0

        for r in range(len(nums)):
            if nums[r] == 1:
                count += 1
                maxCount = max(count, maxCount)
            
            else:
                count = 0
        
        return maxCount
